vm/src/memory/locals.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <errors.h>
+#include <string>
+
+/**
+ * an access to an out of bound index has been made from the `Locals` area
+ */
+class LocalsOutOfBoundError : public MemoryError {
+public:
+    explicit LocalsOutOfBoundError(std::string msg) : MemoryError(std::move(msg)) {}
+};
+
+/**
+ * Encapsulates the allocated locals area of a stack frame
+ */
+class Locals {
+    /**
+     * encapsulated bytes in the stack frame
+     */
+    const char *bytes;
```

Comment:
```suggestion
    char *const bytes;
```
The `const` should apply to the pointer, not the `char`.